core/shared/src/main/scala/result.scala

```diff
@@ -170,6 +170,17 @@ sealed abstract class Result[+T, E <: Exception](val answer: T, val errors: Seq[
       case Unforeseen(e) => throw e
     }
 
+  /** Filter on the answer of this result. */
+  def filter(p: T => Boolean): Result[T, E] =
+    this match {
+      case Errata(_) => this
+      case Answer(b) => if(p(b)) this else Errata[T, E](Nil)
```

Comment:
I don't think `Errata` form a monoid, for the same reason non-empty lists don't. As to the context of the failure, I think that's a choice for the `NotMatchingFilter` exception. I'd be happy for it to have a single parameter denoting the value that failed.